abcpy/ProbabilisticModel.py

```diff
@@ -0,0 +1,719 @@
+from abc import ABCMeta, abstractmethod
+import numpy as np
+
+from scipy.stats import multivariate_normal, norm
+from scipy.special import gamma
+
+#TODO both ricker and lorenz could support that you give some in a combined model, and some not, we should implement that, but not a priority
+
+#TODO docstrings
+
+#NOTE do we maybe want to average over a couple of samples during initialization, rather than taking a single value? this is an issue for example for StudentT!
+
+class ProbabilisticModel(metaclass = ABCMeta):
+    def __init__(self, parameters):
+        self.parents = parameters #all parents of the node
```

Comment:
Why do you call it self.parents and not self.parameters?

Reply:
Because it is not just a parameter, it is a whole probabilistic model. As for the probabilistic model class, it contains more information than just what the parameter is. But I agree that the name is probably not the best, it was more of a placeholder. I am, however, not sure that self.parameters would be the best name.